pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -7817,6 +7817,58 @@ func Test_server_Upsert(t *testing.T) {
 		}
 		return nil
 	}
+
+	/*
+		Upsert test cases (only test float32 unless otherwise specified):
+		- Equivalence Class Testing ( 1000 vectors inserted before a upsert )
+			- case 1.1: success upsert with new ID
+			- case 1.2: success upsert with existent ID
+			- case 2.1: fail upsert with one different dimension vector (type: uint8)
+			- case 2.2: fail upsert with one different dimension vector (type: float32)
+		- Boundary Value Testing ( 1000 vectors inserted before a upsert )
+			- case 1.1: fail upsert with "" as ID
+			- case 2.1: success upsert with ^@ as ID
+			- case 2.2: success upsert with ^I as ID
+			- case 2.3: success upsert with ^J as ID
+			- case 2.4: success upsert with ^M as ID
+			- case 2.5: success upsert with ^[ as ID
+			- case 2.6: success upsert with ^? as ID
+			- case 3.1: success upsert with utf-8 ID from utf-8 index
+			- case 3.2: fail upsert with utf-8 ID from s-jis index
+			- case 3.3: fail upsert with utf-8 ID from euc-jp index
+			- case 3.4: fail upsert with s-jis ID from utf-8 index
+			- case 3.5: success upsert with s-jis ID from s-jis index
+			- case 3.6: fail upsert with s-jis ID from euc-jp index
+			- case 3.4: fail upsert with euc-jp ID from utf-8 index
+			- case 3.5: fail upsert with euc-jp ID from s-jis index
+			- case 3.6: success upsert with euc-jp ID from euc-jp index
+			- case 4.1: success upsert with 😀 as ID
+			- case 5.1: success upsert with one 0 value vector (type: uint8)
+			- case 5.2: success upsert with one +0 value vector (type: float32)
+			- case 5.3: success upsert with one -0 value vector (type: float32)
+			- case 6.1: success upsert with one min value vector (type: uint8)
+			- case 6.2: success upsert with one min value vector (type: float32)
+			- case 7.1: success upsert with one max value vector (type: uint8)
+			- case 7.2: success upsert with one max value vector (type: float32)
+			- case 8.1: success upsert with one NaN value vector (type: float32) // NOTE: To fix it, it is necessarry to check all of vector value
+			- case 9.1: success upsert with one +inf value vector (type: float32)
+			- case 9.2: success upsert with one -inf value vector (type: float32)
+			- case 10.1: fail upsert with one nil vector
+			- case 11.1: fail upsert with one empty vector
+		- Decision Table Testing
```

Comment:
Should we insert vectors before testing same as Boundary Value Testing?

Reply:
add new id testing

Reply:
What I'd like to say is the NGT server status before testing is the same as Boundary testing?
The Boundary testing shows that `1000 vectors inserted before an upsert.`
https://github.com/vdaas/vald/pull/1685/files#diff-059e77835fd022fce77c6202e61ad5a1148830b650e633e31fbee8ab1f824431R7828

Reply:
@kmrmt 
I got it.
let me know when you have updated!